File: 152-maximum-product-subarray/maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int n = nums.size();
        int left = 0, right = 1;
        int res = nums[0];

        for(int i = 0; i < n; i++)
        {
            left = left == 0 ? 1 : left;
            right = right == 0 ? 1 : right;

            left *= nums[i];
            right *= nums[n - 1 - i];

            res = max(res, max(left, right));

        }
        return res;
    }
};

File: 152-maximum-product-subarray/maximum-product-subarray.py
class Solution(object):
    def maxProduct(self, arr):
        """
        :type nums: List[int]
        :rtype: int
        """
        left, right = 1, 1
        ans = arr[0]
		
        for i in range(len(arr)):
		    left = 1 if left == 0 else left
		    right = 1 if right == 0 else right
		    
		    left *= arr[i]
		    
		    right *= arr[len(arr) - 1 - i]
		    
		    ans = max(ans, max(left, right))
        return ans
        